### ISSUE TITLE
Allow comment flags on scope level

### ISSUE THREAD

**Issue by zanona** [eslint/eslint#1830]
Coming from JSLint, I do remember such behaviour was set by default:

```js
function foo() {
   /*global b*/
   more(b);
}

function bar() {
   more(b) // This would trigger an error since b was not defined on this scope
}
```

and the same for cases like:

```js
function foo() {
   /*eslint no-eval:0*/
   var result = eval('1 + 1');
}

function bar() {
   var result2 = eval('1 + 1'); //This would trigger an error since the flag was not set in this scope level.
}
```

However, on eslint every comment declaration  seems to take global effect over the file and not per the scoped it has been applied to?

**Comment by lo1tuma** [comment]
Duplicate of #1766.
